File: backend/app/models/sessions.py
"""セッション関連の API モデル。"""

from datetime import datetime
from typing import Dict

from pydantic import BaseModel, Field


class SessionCreateRequest(BaseModel):
    """セッション作成リクエスト。"""

    server_id: str = Field(..., description="対象となるカタログサーバー ID")
    image: str = Field(..., description="起動する Docker イメージ")
    env: Dict[str, str] = Field(
        default_factory=dict, description="環境変数（Bitwarden 参照を含むことがある）"
    )
    idle_minutes: int = Field(
        default=30,
        ge=1,
        le=240,
        description="アイドルタイムアウト（分）。既定は 30 分。",
    )


class SessionCreateResponse(BaseModel):
    """セッション作成レスポンス。"""

    session_id: str = Field(..., description="生成されたセッション ID")
    container_id: str = Field(..., description="起動したコンテナ ID")
    gateway_endpoint: str = Field(..., description="ゲートウェイ接続先")
    metrics_endpoint: str = Field(..., description="メトリクスエンドポイント")
    idle_deadline: datetime = Field(..., description="アイドル期限 (ISO8601)")

File: frontend/lib/types/sessions.ts
// セッション/実行系の型定義

export interface SessionCreateRequest {
  serverId: string;
  image: string;
  image_digest?: string | null;
  image_thumbprint?: string | null;
  env: Record<string, string>;
  idle_minutes?: number;
}

export interface SessionCreateResponse {
  session_id: string;
  container_id: string;
  gateway_endpoint: string;
  metrics_endpoint: string;
  idle_deadline: string;
}

export interface SessionExecRequest {
  sessionId: string;
  tool: string;
  args: string[];
  asyncMode: boolean;
  maxRunSeconds?: number;
  outputBytesLimit?: number;
}

export interface SessionExecSyncResponse {
  output: string;
  exit_code: number;
  timeout: boolean;
  truncated: boolean;
  started_at: string;
  finished_at: string;
}

export interface SessionExecAsyncResponse {
  job_id: string;
  status: string;
  queued_at: string;
  started_at: string | null;
  result_url: string | null;
}

export interface SessionJobStatusResponse {
  job_id: string;
  status: string;
  output?: string | null;
  exit_code?: number | null;
  timeout: boolean;
  truncated: boolean;
  started_at?: string | null;
  finished_at?: string | null;
}
